forge-gui/res/cardsfolder/upcoming/resourceful_defense.txt

```diff
@@ -0,0 +1,8 @@
+Name:Resourceful Defense
+ManaCost:2 W
+Types:Enchantment
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | ValidCard$ Permanent.YouCtrl+HasCounters | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ Whenever a permanent you control leaves the battlefield, if it had counters on it, put those counters on target permanent you control.
+SVar:TrigPutCounter:DB$ PutCounter | ValidTgts$ Permanent.YouCtrl | TgtPrompt$ Select target permanent you control | CounterType$ EachFromSource | EachFromSource$ TriggeredCardLKICopy
+DeckNeeds:Ability$Counters 
+A:AB$ MoveCounter | Cost$ 4 W | ValidSource$ Permanent.YouCtrl | ValidTgts$ Permanent.YouCtrl | CounterType$ All | CounterNum$ Any | Defined$ Targeted | SpellDescription$ Move any number of counters from target permanent you control to another target permanent you control.

```

Comment:
I'm sad to say this because I like clean one line abilities, but this needs a pump target line to make sure we are targeting both permanents. This is similar to what many Fight cards currently use. This will make sure that triggers that happen when a certain creature is targeted go off appropriately. We also need to make sure they are unique targets because of the phrase "to **another** target permanent"

have a look at `Nesting Grounds`

Will be interesting to see what the AI does with this @Agetian

Reply:
I updated MoveCounter once to have a special case when using exactly 2 targets:

Check out Simic Guildmage:
`A:AB$ MoveCounter | Cost$ 1 G | ValidTgts$ Creature | TargetMin$ 2 | TargetMax$ 2 | TgtPrompt$ Select target creatures to move +1/+1 counters | TargetsWithSameController$ True | CounterType$ P1P1 | CounterNum$ 1 | StackDescription$ SpellDescription | SpellDescription$ Move a +1/+1 counter from target creature onto another target creature with the same controller.`

Reply:
Nesting Grounds take on it works WAY better, thanks!

Reply:
@Northmoc @Agetian @TheBlackMarvel i would prever for MoveCounter with 2 Targets to do it similar to Simic Guildmage using `TargetMin$ 2 | TargetMax$ 2`

Reply:
OK, added.

Reply:
no i mean you can remove the pump there

Reply:
Oooooooh